deps/src/polymake_functions.cpp

```diff
@@ -4,11 +4,16 @@
 #include "polymake_functions.h"
 
 void initialize_polymake(){
-    data.main_polymake_session = new polymake::Main;
-    data.main_polymake_session->shell_enable();
-    data.main_polymake_scope = new polymake::perl::Scope(data.main_polymake_session->newScope());
-
-    std::cout << data.main_polymake_session->greeting() << std::endl;
+    try {
+        if (data.main_polymake_session == nullptr) {
```

Comment:
Simple nitpick, even if this merged already: If we get out of this function too early, `main_polymake_session` might be set, but `polymake_scope` not. Maybe check both pointers?